src/python/foglamp/device/http_south_device.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+"""HTTP Listener handler for sensor readings"""
+
+from aiohttp import web
+import asyncio
+from foglamp import logger
+from foglamp.web import middleware
+from foglamp.device.ingest import Ingest
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_LOGGER = logger.setup(__name__)
+
+_CONFIG_CATEGORY_NAME = 'http_south'
+_CONFIG_CATEGORY_DESCRIPTION = 'South Plugin HTTP Listener'
+_DEFAULT_CONFIG = {
+    'plugin': {
+         'description': 'http_south_device',
+         'type': 'string',
+         'default': 'http_south'
+    },
+    'port': {
+        'description': 'Port to listen on',
+        'type': 'integer',
+        'default': '6683',
+    },
+    'uri': {
+        'description': 'URI to accept data on',
+        'type': 'string',
+        'default': '0.0.0.0',
+    }
+}
+
+
+def plugin_info():
+    return {'name': 'http_south', 'version': '1.0', 'mode': 'async', 'type': 'device',
+            'interface': '1.0', 'config': _DEFAULT_CONFIG}
+
+
+def plugin_init(config):
+    """Registers HTTP Listener handler to accept sensor readings"""
+
+    _LOGGER.info("Retrieve HTTP Listener Configuration %s", config)
+
+    uri = config['uri']['value']
+    port = config['port']['value']
+
+    loop = asyncio.get_event_loop()
+
+    app = web.Application(middlewares=[middleware.error_middleware])
+    app.router.add_route('POST', '/', HttpSouthIngest.render_post)
+    handler = app.make_handler()
+    coro = loop.create_server(handler, uri, port)
+
+    return coro, handler
+
+
+def plugin_run(data):
+    pass
+
+def plugin_reconfigure(config):
+    pass
```

Comment:
should we add TODO that it will be called on configuration manager callback run?